### PR TITLE
AWS ruby SDK v2 upgrade

### LINKED ISSUE #11: Switch to aws-sdk-v2
We'll get performance gains and first-class library support from a switch to aws-sdk version 2: https://aws.amazon.com/sdk-for-ruby/

### PR DESCRIPTION
**Pull request by dwagner2301**
API-Calls changed to support the AWS ruby SDK v2. Unsupported config
parameters and exceptions removed.

fixes #11

### CODE REVIEW

**Comment by dwagner2301** [comment]
@ph  formal tests are still missing but I'm happy to provide those if somebody points me into the right direction to get it done.

**Comment by ph** [comment]
I think the easiest way to make to write the test would be to do an integration test.
What we could do is to use the `logstash-output-sqs` and push events to an SQS queue and get them with the `logstash-input-sqs` and compare them.

I've done something similar for the S3 output at https://github.com/logstash-plugins/logstash-output-s3/blob/master/spec/integration/s3_spec.rb

I make sure all the credentials are in the environment.

What do you think?

**Comment by ph** [comment]
I can help tomorrow with the test and show you what I have done.

File: logstash-input-sqs.gemspec [hunk 1]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency "logstash-core", '>= 1.4.0', '< 2.0.0'
 
   s.add_runtime_dependency 'logstash-codec-json'
-  s.add_runtime_dependency 'aws-sdk'
+  s.add_runtime_dependency 'aws-sdk', '~> 2.1.0'
```
Comment (ph) [inline logstash-input-sqs.gemspec]: We don't need to depend directly on aws-sdk v2.
We can just define a dependencie on logstash-mixin-aws, like this:

``` ruby
  s.add_runtime_dependency "logstash-mixin-aws", ">= 1.0.0"
```

File: lib/logstash/inputs/sqs.rb [hunk 2]
```diff
@@ -137,34 +138,27 @@ def run(output_queue)
   end # def run
 
   def teardown
-    @sqs_queue = nil
+    @sqs = nil
```
Comment (ph) [inline lib/logstash/inputs/sqs.rb]: I think explicitly doing isn't needed, once the plugin isn't used the GC will clear the memory.

**Comment by dwagner2301** [comment]
@ph I just added your suggested changes to both plugins. To implement the tests the way you did it with the S3 plugin looks doable. But I won't have a chance to start with that before the weekend. Now, I do understand how to write the tests but I'm still not sure how to run them.

**Comment by dwagner2301** [comment]
@ph I got a dummy test working. I might be able to work on the implementation of the actual integration test by following the S3 example during the next week. Let me know, if you need it done by a specific date and I will try to squeeze it in.

**Comment by ph** [comment]
@dwagner2301 If you don't mind, I'll fork your PR and add test to it?

**Comment by dwagner2301** [comment]
@ph I absolutely don't mind! I would have loved to to implement the test but right now it seems you implementing it seems like the best way to get it done as soon as possible.

**Comment by ph** [comment]
I am closing this in favor of https://github.com/logstash-plugins/logstash-input-sqs/pull/18

**Pull request closed without merging**